src/custom/state/price/updater.ts

```diff
@@ -156,17 +153,20 @@ export default function FeesUpdater(): null {
       const quoteParams = { buyToken, chainId, sellToken, kind, amount: amount.raw.toString() }
 
       // if no token is unsupported and needs refetching
-      const refetchAll = !unsupportedToken && isRefetchQuoteRequired(quoteParams, quoteInfo)
-      const refetchPrice = !unsupportedToken && priceIsOld(quoteInfo)
+      const hasToRefetch = !unsupportedToken && isRefetchQuoteRequired(isLoading, quoteParams, quoteInfo)
 
-      if (unsupportedNeedsCheck || refetchAll || refetchPrice) {
-        const shouldFetchNewQuote = (quoteInfo && !quoteUsingSameParameters(quoteParams, quoteInfo)) || true
+      if (unsupportedNeedsCheck || hasToRefetch) {
+        // Decide if this is a new quote, or just a refresh
+        const thereIsPreviousPrice = !!quoteInfo?.price?.amount
+        const isJustPriceRefresh = quoteInfo
+          ? thereIsPreviousPrice && quoteUsingSameParameters(quoteParams, quoteInfo)
+          : false
 
         refetchQuote({
           quoteParams,
-          fetchFee: refetchAll,
+          fetchFee: true, // TODO: Review this, because probably now doesn't make any sense to not query the feee in some situations. Actually the endpoint will change to one that returns fee and quote together
```

Comment:
We will need to revisit this part. 
Although for this PR should be fine. We always query the fee.